plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -1854,16 +1855,23 @@ producer_plugin_impl::start_block_result producer_plugin_impl::start_block() {
       }
 
       try {
+         _account_fails.clear();
+
+         if( _pending_block_mode == pending_block_mode::speculating && _received_block )
+            return start_block_result::exhausted;
          if( !remove_expired_trxs( preprocess_deadline ) )
             return start_block_result::exhausted;
+         if( _pending_block_mode == pending_block_mode::speculating && _received_block )
+            return start_block_result::exhausted;
          if( !remove_expired_blacklisted_trxs( preprocess_deadline ) )
             return start_block_result::exhausted;
+         if( _pending_block_mode == pending_block_mode::speculating && _received_block )
+            return start_block_result::exhausted;
          if( !_subjective_billing.remove_expired( _log, chain.pending_block_time(), fc::time_point::now(), preprocess_deadline ) )
             return start_block_result::exhausted;
```

Comment:
```suggestion
         if(should_stop_speculating() || !remove_expired_trxs( preprocess_deadline ) )
            return start_block_result::exhausted;
         if(should_stop_speculating() || ! !remove_expired_blacklisted_trxs( preprocess_deadline ) )
            return start_block_result::exhausted;
         if(should_stop_speculating() || ! !_subjective_billing.remove_expired( _log, chain.pending_block_time(), fc::time_point::now(), preprocess_deadline ) )
            return start_block_result::exhausted;
```

Reply:
Super sorry, I commited my suggestion by mistake! I didn't realize that was what "Commit suggestion" did!.
Should I try to undo it?

Reply:
I reverted this commit.